src/main/java/com/google/firebase/auth/SamlProviderConfig.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.auth;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.firebase.auth.ProviderConfig.AbstractCreateRequest;
+import com.google.firebase.auth.ProviderConfig.AbstractUpdateRequest;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Contains metadata associated with a SAML Auth provider.
+ *
+ * <p>Instances of this class are immutable and thread safe.
+ */
+public final class SamlProviderConfig extends ProviderConfig {
+
+  @Key("idpConfig")
+  private IdpConfig idpConfig;
+
+  @Key("spConfig")
+  private SpConfig spConfig;
+
+  public String getIdpEntityId() {
+    return idpConfig.getIdpEntityId();
+  }
+
+  public String getSsoUrl() {
+    return idpConfig.getSsoUrl();
+  }
+
+  public List<String> getX509Certificates() {
+    ImmutableList.Builder<String> certificates = ImmutableList.<String>builder();
+    for (IdpCertificate idpCertificate : idpConfig.getIdpCertificates()) {
+      certificates.add(idpCertificate.getX509Certificate());
+    }
+    return certificates.build();
+  }
+
+  public String getRpEntityId() {
+    return spConfig.getRpEntityId();
+  }
+
+  public String getCallbackUrl() {
+    return spConfig.getCallbackUrl();
+  }
+
+  static List<Object> ensureNestedList(Map<String, Object> outerMap, String id) {
+    List<Object> list = (List<Object>) outerMap.get(id);
```

Comment:
I think we are missing some imports here causing a compilation error.

Reply:
Done.